mobile_packet_verifier/src/burner.rs

```diff
@@ -130,17 +130,55 @@ where
                     .await?;
             }
         }
-
         Ok(())
     }
 
     async fn burn_data_credits(
         &self,
         payer: &PublicKeyBinary,
         amount: u64,
-    ) -> Result<(), S::Error> {
-        let txn = self.solana.make_burn_transaction(payer, amount).await?;
-        self.solana.submit_transaction(&txn).await?;
+    ) -> Result<(), BurnError<S::Error>> {
+        let txn = self
+            .solana
+            .make_burn_transaction(payer, amount)
+            .await
+            .map_err(BurnError::SolanaError)?;
+        let mut signed_txn = self
+            .solana
+            .sign_transaction(&txn)
+            .await
+            .map_err(BurnError::SolanaError)?;
+        // handle retries, if we encounter a blockhash not found error
+        // resign the txn with the latest blockhash before next retry attempt
+        let mut attempt = 1;
+        const MAX_ATTEMPTS: u64 = 10;
+        loop {
+            match self.solana.submit_transaction(&signed_txn).await {
+                Ok(_) => {
+                    tracing::info!(%payer, %amount, "Burned DC");
+                    break;
+                }
+                Err(err) if err.is_error_blockhash_not_found() && attempt < MAX_ATTEMPTS => {
+                    tracing::error!(%payer, %amount, "block hash not found..possibly stale block hash, resigning txn and retrying");
+                    signed_txn = self
+                        .solana
+                        .sign_transaction(&txn)
+                        .await
+                        .map_err(BurnError::SolanaError)?;
+
+                    attempt += 1;
+                    continue;
+                }
+                Err(_) if attempt < MAX_ATTEMPTS => {
+                    attempt += 1;
+                    tokio::time::sleep(Duration::from_secs(attempt)).await;
```

Comment:
The boost_manager backs off of a retry_delay with the attempt. Should that also be happening here and in the iot_packet_verifier?

Reply:
yeah it should have been, i will add that in

Reply:
addressed here: cdd5a3263e48e0b03ec08c7bdd438babda374fc2